catalog/app/containers/Bucket/PackageUpdateDialog.js

```diff
@@ -0,0 +1,933 @@
+    overflow: 'hidden',
+  },
+  dropzoneErr: {
+    borderColor: t.palette.error.main,
+  },
+  dropzoneWarn: {
+    borderColor: t.palette.warning.dark,
+  },
+  active: {
+    background: t.palette.action.selected,
+  },
+  dropMsg: {
+    ...t.typography.body2,
+    alignItems: 'center',
+    display: 'flex',
+    flexGrow: 1,
+    justifyContent: 'center',
+    paddingBottom: t.spacing(1),
+    paddingTop: t.spacing(1),
+    textAlign: 'center',
+  },
+  dropMsgErr: {
+    color: t.palette.error.main,
+  },
+  dropMsgWarn: {
+    color: t.palette.warning.dark,
+  },
+  filesContainer: {
+    borderBottom: `1px solid ${t.palette.action.disabled}`,
+    maxHeight: 200,
+    overflowX: 'hidden',
+    overflowY: 'auto',
+  },
+  filesContainerErr: {
+    borderColor: t.palette.error.main,
+  },
+  filesContainerWarn: {
+    borderColor: t.palette.warning.dark,
+  },
+  added: {},
+  modified: {},
+  deleted: {},
+  unchanged: {},
+  fileEntry: {
+    alignItems: 'center',
+    background: t.palette.background.paper,
+    display: 'flex',
+    '&:not(:last-child)': {
+      borderBottomStyle: 'solid',
+      borderBottomWidth: '1px',
+      borderColor: 'inherit',
+    },
+    '&$added': {
+      background: M.colors.green[100],
+    },
+    '&$modified': {
+      background: M.colors.yellow[100],
+    },
+    '&$deleted': {
+      background: M.colors.red[100],
+    },
+  },
+  filePath: {
+    ...t.typography.body2,
+    flexGrow: 1,
+    marginRight: t.spacing(1),
+    overflow: 'hidden',
+    textOverflow: 'ellipsis',
+    whiteSpace: 'nowrap',
+  },
+  fileSize: {
+    ...t.typography.body2,
+    color: t.palette.text.secondary,
+    marginRight: t.spacing(0.5),
+  },
+  lock: {
+    alignItems: 'center',
+    background: 'rgba(255,255,255,0.9)',
+    border: `1px solid ${t.palette.action.disabled}`,
+    borderRadius: t.shape.borderRadius,
+    bottom: 0,
+    cursor: 'not-allowed',
+    display: 'flex',
+    flexDirection: 'column',
+    justifyContent: 'center',
+    left: 0,
+    position: 'absolute',
+    right: 0,
+    top: 0,
+  },
+  progressContainer: {
+    display: 'flex',
+    position: 'relative',
+  },
+  progressPercent: {
+    ...t.typography.h5,
+    alignItems: 'center',
+    bottom: 0,
+    display: 'flex',
+    justifyContent: 'center',
+    left: 0,
+    position: 'absolute',
+    right: 0,
+    top: 0,
+  },
+  progressSize: {
+    ...t.typography.body2,
+    color: t.palette.text.secondary,
+    marginTop: t.spacing(1),
+  },
+}))
+
+function FilesInput({ input, meta, uploads, setUploads, errors = {} }) {
+  const classes = useFilesInputStyles()
+
+  const value = input.value || { added: {}, deleted: {}, existing: {} }
+  const disabled = meta.submitting || meta.submitSucceeded
+  const error = meta.submitFailed && meta.error
+
+  const onInputChange = input.onChange
+
+  const totalSize = React.useMemo(
+    () => Object.values(value.added).reduce((sum, f) => sum + f.size, 0),
+    [value.added],
+  )
```

Comment:
`inputValue` in `PackageCreateDialog` and `{ added, deleted, existing }` in `PackageUploadDialog` have different data format, based on `totalSize` function. Maybe things would be simpler if they have the same data format

Reply:
well, these dialogs use different data and slightly different display for the files input, so i dont think we can make things simpler by somehow converting different data to the same format, it doesnt make sense (correct me if i dont get your comment right).
however, we may somehow unify the files input components and parameterize the resulting component to have different display for plain file upload (package creation) vs diff-like file list editing (package update), but this will actually make things more complex, not simpler (in Rich Hicky sense).
i'm open to further refactoring the dialogs to reuse more code in some near future.